[WFLY-15524]: Upgrade netty from 4.1.68 to 4.1.70.

ehsavoie opened the pull request:
* Upgrading netty to 4.1.70.
* Removing netty-all and using the proper netty artefacts in pom.xml.
* Removing the io.netty jboss module and creating io.netty.netty-*
  modules for each artefact.
* Adding a task to manage the native parts.

Jira: https://issues.redhat.com/projects/WFLY/issues/WFLY-15524
Requires: https://github.com/wildfly/wildfly/pull/14886

bstansberry commented on ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-epoll/pm/wildfly/tasks.xml:
```diff
@@ -0,0 +1,13 @@
+        <filter pattern="META-INF/" include="true" />
+        <filter pattern="META-INF/native/*" include="true" />
+        <filter pattern="*" include="false"/>
+    </copy-artifact>-->
```
@ehsavoie Looks like this is pure cruft as it seems to be a commented out copy of the uncommented block.

ehsavoie replied:
different classifier : linux-aarch_64

bstansberry commented:
@pferraro @kabir Please review

kabir commented on ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-epoll/pm/wildfly/tasks.xml:
```diff
@@ -0,0 +1,13 @@
+        <filter pattern="META-INF/native/*" include="true" />
+        <filter pattern="*" include="false"/>
+    </copy-artifact>
+<!--    <copy-artifact artifact="io.netty:netty-transport-native-epoll::linux-aarch_64" to-location="modules/system/layers/base/io/netty/netty-transport-native-epoll/main/lib" extract="true" >
```
Cruft?

kabir commented on ee-feature-pack/galleon-common/src/main/resources/packages/io.netty.netty-transport-native-unix-common/pm/wildfly/tasks.xml:
```diff
@@ -0,0 +1,15 @@
+        <filter pattern="*" include="false"/>
+    </copy-artifact>
+    <!--
+    <copy-artifact artifact="io.netty:netty-transport-native-epoll::linux-aarch_64" to-location="modules/system/layers/base/io/netty/netty-transport-native-epoll/main/lib" extract="true" >
```
Cruft?

ehsavoie replied:
nope i was wondering about adding aarch_64 binaries

kabir commented on microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/vertx/core/main/module.xml:
```diff
@@ -33,7 +33,21 @@
         <module name="com.fasterxml.jackson.core.jackson-core"/>
         <module name="com.fasterxml.jackson.core.jackson-databind"/>
-        <module name="io.netty"/>
+        <!--<module name="io.netty"/>-->
```
Cruft?

ehsavoie replied:
yes

kabir reviewed (dismissed):
Some small comments aside, it looks good to me. I've restarted the two jobs that were failing.
**Linux SM JDK8** - had a failure in IIOP probably not related
**WildFly Preview** - had a lot of failures, hopefully just one of those bad runs

boris-unckel commented on microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-resolver-dns/main/module.xml:
```diff
@@ -0,0 +1,51 @@
+    </resources>
+
+    <dependencies>
+        <module name="java.logging"/>
+        <module name="java.management"/>
```
Wondering why I did not see these module.xml in the first round checking. Only java.naming is needed here.
java.logging, java.management can be removed.

boris-unckel commented on microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-resolver-dns/main/module.xml:
```diff
@@ -0,0 +1,51 @@
+        <module name="java.logging"/>
+        <module name="java.management"/>
+        <module name="java.naming"/>
+        <module name="java.xml"/>
+        <!--WFLY-14219 Remove deprecated <module name="javax.api"/> -->
+        <module name="sun.jdk"/>
+        <module name="jdk.sctp"/>
+        <module name="jdk.unsupported"/>
```
java.xml, sun.jdk, jdk.sctp, jdk.unsupported can be removed.

boris-unckel commented on microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/netty/netty-codec-http2/main/module.xml:
```diff
@@ -0,0 +1,51 @@
+    </resources>
+
+    <dependencies>
+        <module name="java.logging"/>
+        <module name="java.management"/>
+        <module name="java.naming"/>
+        <module name="java.xml"/>
+        <!--WFLY-14219 Remove deprecated <module name="javax.api"/> -->
+        <module name="sun.jdk"/>
+        <module name="jdk.sctp"/>
+        <module name="jdk.unsupported"/>
```
java.logging, java.management, java.naming, java.xml, sun.jdk, jdk.sctp, jdk.unsupported can be removed

kabir commented:
@wildfly/prod test

kabir commented:
@ehsavoie the WF Preview failures seem related

darranl commented:
/retest

darranl commented:
@ehsavoie FYI this one has a merge conflict.

ehsavoie committed:
[WFLY-15524]: Upgrade netty from 4.1.68 to 4.1.72.

* Upgrading netty to 4.1.72.
* Removing netty-all and using the proper netty artefacts in pom.xml.
* Removing the io.netty jboss module and creating io.netty.netty-*
  modules for each artefact.
* Adding a task to manage the native parts.

Jira: https://issues.redhat.com/projects/WFLY/issues/WFLY-15524

ehsavoie commented:
@darranl should be ok now

darranl commented:
@kabir could you please have another look, can we dismiss your review?

kabir commented:
@darranl done

bstansberry commented:
Thanks, @ehsavoie
